src/batch.js

```diff
@@ -1,28 +1,221 @@
 'use strict'
 
+const router = require('lambda-router/src/router')
+const { expectation } = require('sinon')
+
 const { HttpError } = require('./httpError')
 
 const batchDefaultConfig = {
-  maxBatchSize: 20
+  maxBatchSize: 20,
+  handler: undefined
```

Comment:
handler allows a consumer to bypass the flow:
event -> batchRoute -> hander+handler+handler

Instead allowing them to customize the behavior of each called handler, or provide a way to run each though the initial event flow. For example:
eventInit -> batchRoute -> eventInit  -> lambda-route -> handler
or
eventInit -> batchRoute -> customAuthorizer -> customRoute -> handler

Reply:
There are two cases here.

1. The app wants to run a function before the entire batch
2. The app wants to run a function before each route

Both of these are easily achievable. (1) can be done by running the function before the call to `route.route()`, (2) can be done by providing a wrapped handler for the route.

Adding another injection point is unnecessary and confusing.

Reply:
I think that's reasonable, but I still worry about a situation such as:

On event, The app initializes a logger, assigns a uniqueId to the request, and then calls router.route().
The desire is to assign a sub-id to each batched request, but there's not a way to supply this to the batch route, nor is there a good indication from the wrapped handler as to it being part of a batched request.


Adding a flag or identifier to the context or event to indicate it's part of a batched execution would likely accomplish the same goal however.

Reply:
I agree that adding data to the event to indicate it is part of a batch request is a good idea. As for adding a sub-id to each route: this can be done inside the route, even conditionally on the `event.batchRequestId` flag (or whatever we go with).